babybuddy/middleware.py

```diff
@@ -158,6 +145,82 @@ def __call__(self, request: HttpRequest):
 
-        return self.get_response(request)
+        if apply_x_ingress_path:
+            set_script_prefix("/" + x_ingress_path.lstrip("/"))
+        else:
+            set_script_prefix(self.original_script_prefix)
+
+        response = self.get_response(request)
+
+        if apply_x_ingress_path:
+            is_redirect_response = isinstance(
+                response, HttpResponseRedirect
+            ) or response.status_code in [301, 307, 308]
+            if is_redirect_response:
+                split_url = urlsplit(response["Location"])
+                path_prefix = "/" + x_ingress_path.lstrip("/")
+                if not split_url.path.startswith(path_prefix):
+                    new_url = urlunsplit(
+                        (
+                            split_url.scheme,
+                            split_url.netloc,
+                            "/" + x_ingress_path.lstrip("/") + split_url.path,
+                            split_url.query,
+                            split_url.fragment,
+                        )
+                    )
+                    response["Location"] = new_url
+            elif isinstance(response, StreamingHttpResponse):
+                # Pray that the response works
+                logging.error(
+                    "HomeAssistant middleware: StreamingHttpResponse is not "
+                    "supported. Resulting URLs to home assistant ingress might "
+                    "be incorrect."
+                )
+            elif isinstance(response, HttpResponse):
+                if response["Content-Type"].lower().startswith("text/html"):
+                    # Filter /static and /media URLs, I did not find a better
+                    # way that would be compatible with external third-party apps.
+                    content = response.content.decode()
+                    static_trunc = settings.STATIC_URL.rstrip("/")
+                    media_trunc = settings.MEDIA_URL.rstrip("/")
+
+                    content = (
+                        content.replace(
+                            f'"{static_trunc}',
+                            f'"{x_ingress_path}{static_trunc}',
+                        )
+                        .replace(
+                            f"'{static_trunc}",
+                            f"'{x_ingress_path}{static_trunc}",
+                        )
+                        .replace(
+                            f'"{media_trunc}',
+                            f'"{x_ingress_path}{media_trunc}',
+                        )
+                        .replace(
+                            f"'{media_trunc}",
+                            f"'{x_ingress_path}{media_trunc}",
+                        )
+                    )
+                    filtered_headers = {
+                        key: value
+                        for key, value in response.headers.items()
+                        if not key.lower().startswith("content-")
+                    }
```

Comment:
This part is necessary to make media and static urls work. There are different solutions for this, in theory, but plugins do not like them. So instead, filtering the returned text for any url encountered and fixing up the path, seems to be the only "scalable" way... 

So this is pretty much a re-implementation of the pattern replace-rules from the home assistant integration's nginx-rules.